src/authorizations/components/AllAccessTokenOverlay.tsx

```diff
@@ -19,122 +20,98 @@ import {
+import {Authorization} from 'src/types'
 
 interface OwnProps {
   onClose: () => void
 }
 
-interface State {
-  description: string
-}
-
-type ReduxProps = ConnectedProps<typeof connector>
-type Props = OwnProps & ReduxProps
-
-@ErrorHandling
-class AllAccessTokenOverlay extends PureComponent<Props, State> {
-  public state = {description: ''}
-
-  render() {
-    const {description} = this.state
-
-    return (
-      <Overlay.Container maxWidth={500}>
-        <Overlay.Header
-          title="Generate All Access API Token"
-          onDismiss={this.handleDismiss}
-        />
-        <Overlay.Body>
-          <Form onSubmit={this.handleSave}>
-            <FlexBox
-              alignItems={AlignItems.Center}
-              direction={FlexDirection.Column}
-              margin={ComponentSize.Large}
-            >
-              <Alert
-                icon={IconFont.AlertTriangle}
-                color={ComponentColor.Warning}
-              >
-                This token will be able to create, update, delete, read, and
-                write to anything in this organization
-              </Alert>
-              <Form.Element label="Description">
-                <Input
-                  placeholder="Describe this new API token"
-                  value={description}
-                  onChange={this.handleInputChange}
-                  testID="all-access-token-input"
-                />
-              </Form.Element>
-
-              <Form.Footer>
-                <Button
-                  text="Cancel"
-                  icon={IconFont.Remove}
-                  onClick={this.handleDismiss}
-                />
-
-                <Button
-                  text="Save"
-                  testID="button--save"
-                  icon={IconFont.Checkmark}
-                  color={ComponentColor.Success}
-                  type={ButtonType.Submit}
-                />
-              </Form.Footer>
-            </FlexBox>
-          </Form>
-        </Overlay.Body>
-      </Overlay.Container>
-    )
-  }
-
-  private handleSave = () => {
-    const {orgID, meID, onCreateAuthorization} = this.props
+const AllAccessTokenOverlay: FC<OwnProps> = props => {
+  const {onClose} = useContext(OverlayContext)
+  const dispatch = useDispatch()
+  const [description, setDescription] = useState('')
+  const {id: orgID} = useSelector(getOrg)
+  const {id: meID} = useSelector(getMe)
 
+  const handleSave = () => {
     const token: Authorization = {
       orgID,
-      description: this.state.description,
+      description: description,
       permissions: allAccessPermissions(orgID, meID),
     }
-    onCreateAuthorization(token)
-
-    this.handleDismiss()
+    dispatch(createAuthorization(token))
+    handleDismiss()
+    dispatch(showOverlay('access-token', null, () => dismissOverlay()))
   }
 
-  private handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+  const handleInputChange = e => {
```

Comment:
isn't this still a `ChangeEvent<HTMLInputElement>`?

also can you rename this `event` or `evt` please? `e` can stand for `event` or `error`, and `e` is a bit overloaded when doing global searches